src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -567,4 +571,64 @@ public virtual void PrintMessageNormalized(TextWriter writer, string message, bo
         else if (!removeLastNewLine)
             writer.WriteLine();
     }
+
+    /// <summary>
+    /// Adds a new runtime option. Throws if option already exists.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    public void AddOption<T>(string name, T value)
+    {
+        Options.Add(name, value);
+    }
+
+    /// <summary>
+    /// Trys to add a new runtime option.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="name"></param>
+    /// <param name="value"></param>
+    /// <returns>True if successful. False otherwise.</returns>
+    public bool TryAddOption<T>(string name, T value)
+    {
+        if (Options.ContainsKey(name))
+            return false;
+        AddOption<T>(name, value);
+        return true;
+    }
+
+    public void AddOrOverwriteOption<T>(string name, T value)
+    {
+        if (!Options.ContainsKey(name))
+            AddOption<T>(name, value);
+        else
+            Options[name] = value;
```

Comment:
Can't you just do `Options[name] = value`. You don't have to check if it exist or not.

Reply:
Fixed.